instrumentation/mongo/lib/opentelemetry/instrumentation/mongo/middlewares/subscriber.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative 'command_serializer'
+
+module OpenTelemetry
+  module Instrumentation
+    module Mongo
+      module Middlewares
+        # Event handler class for Mongo Ruby driver
+        class Subscriber
+          THREAD_KEY = :__opentelemetry_mongo_spans__
+
+          def started(event) # rubocop:disable Metrics/AbcSize
+            # start a trace and store it in the current thread; using the `operation_id`
+            # is safe since it's a unique id used to link events together. Also only one
+            # thread is involved in this execution so thread-local storage should be safe. Reference:
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring.rb#L70
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring/publishable.rb#L38-L56
+            # TODO: decide naming convention
+            name = 'mongo.cmd'
+            # name = "#{event.database_name}.#{event.command_name}"
+            span = tracer.start_span(name, kind: :client)
+            set_span(event, span)
+
+            serialized_command = CommandSerializer.new(event.command).serialize
+            collection = get_collection(event.command)
+            span.set_attribute('component', 'mongo-ruby')
+            span.set_attribute('db.type', 'mongodb')
+            span.set_attribute('db.instance', event.database_name)
+            span.set_attribute('mongo.command', event.command_name)
+            span.set_attribute('mongo.collection', collection) if collection
+            span.set_attribute('db.statement', serialized_command)
+            span.set_attribute('mongo.request_id', event.request_id)
+            span.set_attribute('mongo.op_id', event.operation_id)
```

Comment:
```suggestion
```
These are not part of the OTel semantic conventions for Mongo: https://github.com/open-telemetry/opentelemetry-specification/blob/323cd1d8811d3e3df08bec689f9dacd6cb782888/semantic_conventions/trace/database.yaml#L283-L295

Reply:
I think they should be... these will make it much easier to trace with respect to changes in the database, especially op_id which would allow the trace to be linked to the opLog

Reply:
Could you please raise this as an issue (or open a PR) in the spec repo? I suspect they should be `db.mongodb.request_id` and `db.mongodb.op_id`, based on existing semantic conventions.